### PR TITLE
CBL-5933: Replicator log contains auth info

### PR DESCRIPTION
**Pull request by jianminzhao**
Made a white list based on the current constants of keys of repl::Options except for kC4ReplicatorAuthPassword. We only log properties in the white list.

Platform added options won't be logged because of the white-list.

### CODE REVIEW

File: Replicator/Worker.cc [hunk 1]
```diff
@@ -336,4 +335,71 @@ namespace litecore::repl {
...
+            // [1]: Auth dictionary keys:
+            kC4ReplicatorAuthType,
+            kC4ReplicatorAuthUserName,
+            // kC4ReplicatorAuthPassword,
+            kC4ReplicatorAuthEnableChallengeAuth,
+            kC4ReplicatorAuthClientCert,
+            kC4ReplicatorAuthClientCertKey,
+            kC4ReplicatorAuthToken,
+
+            // [2]: auth.type values:
```
Comment (borrrden) [inline Replicator/Worker.cc:386]: These are values, not keys, and thus don't need to be considered

File: Replicator/Worker.cc [hunk 2]
```diff
@@ -336,4 +335,71 @@ namespace litecore::repl {
...
+            kC4AuthTypeClientCert,
+
+            // [3]: Proxy dictionary keys:
+            kC4ReplicatorProxyType,
+            kC4ReplicatorProxyHost,
+            kC4ReplicatorProxyPort,
+            kC4ReplicatorProxyAuth,
+
+            // [4]: proxy.type values:
+            kC4ProxyTypeNone,
```
Comment (borrrden) [inline Replicator/Worker.cc:400]: Same comment

**Review by borrrden — CHANGES_REQUESTED** [review]

File: Replicator/Worker.cc (hunk 1 above)
Comment (jianminzhao) [reply Replicator/Worker.cc:386]: good point

**Review by borrrden — APPROVED** [review]

File: Replicator/Worker.cc [hunk 3]
```diff
@@ -336,4 +335,58 @@ namespace litecore::repl {
...
+            kC4ReplicatorCompressionLevel,
+
+            // [1]: Auth dictionary keys:
+            kC4ReplicatorAuthType,
+            kC4ReplicatorAuthUserName,
+            // kC4ReplicatorAuthPassword,
+            kC4ReplicatorAuthEnableChallengeAuth,
+            kC4ReplicatorAuthClientCert,
+            kC4ReplicatorAuthClientCertKey,
+            kC4ReplicatorAuthToken,
```
Comment (pasin) [inline Replicator/Worker.cc:384]: Shall we log this?

Comment (jianminzhao) [reply Replicator/Worker.cc:384]: When find it out it's not okay to log, we can take it out from the white list easily.

Comment (pasin) [reply Replicator/Worker.cc:384]: This info shouldn't be logged.

Comment (jianminzhao) [reply Replicator/Worker.cc:384]: okay, will remove it.

**Pull request merged**